docs/content/docs/configuration.md

```diff
@@ -51,6 +54,118 @@ status:
 default_decision: /http/example/authz/allow
 ```
 
+## Environment Variable Substitution
+> Only supported with the OPA runtime (`opa run`).
+
+Environment variables referenced with the `${...}` notation within the configuration
+will be replaced with the value of the environment variable.
+
+Example using `BASE_URL` and `BEARER_TOKEN` environment variables:
+```yaml
+services:
+  - name: acmecorp
+    url: ${BASE_URL}
+    credentials:
+      bearer:
+        token: "${BEARER_TOKEN}"
+
+discovery:
+  name: /example/discovery
+  prefix: configuration
+```
+The environment variables `BASE_URL` and `BEARER_TOKEN` will be substituted in when the config
+file is loaded by the OPA runtime.
+
+> If the variable is undefined then an empty string (`""`) is substituted. It will __not__
+raise an error.*
+
+## CLI Runtime Overrides
+> Only supported with the OPA runtime (`opa run`).
+
+Using `opa run` there are CLI options to explicitly set config values. These will override
+any values set in the config file.
+
+There are two options to use: `--set` and `--set-file`
+
+Both options take options in a `key=value` format where they key is a selector for the yaml
+config structure, for example: `decision_logs.reporting.min_delay_seconds=300` is equivelent
+to JSON `{"decision_logs: {"reporting": {"min_delay_seconds: 300}}}`. Multiple values can be
+specified with comma seperators (`key=value,key2=value2,..`). Or with additonal `--set`
+parameters.
+
+Example using several different options:
+```
+opa run \
+  --set "default_decision=/http/example/authz/allow" \
+  --set "services[0].name=acmecorp" \
+  --set "services[0].url=https://test-env/control-plane-api/v1" \
+  --set "services[0].credentials.bearer.token=\${TOKEN}"
+  --set "labels.app=myapp,labels.region=west"
+```
+This is equivelent to a YAML config file that looks like:
+
+```yaml
+services:
+  - name: acmecorp
+    url: https://test-env/control-plane-api/v1
+    credentials:
+      bearer:
+        token: ${TOKEN}
+
+labels:
+  app: myapp
+  region: west
+
+default_decision: /http/example/authz/allow
+```
+
+The `--set-file` option is expecting a file path for the value. This allows keeping secrets in
+files and loading them into the config at run time. For Example:
+
+With a file `/var/run/secrets/bearer_token.txt` that has contents:
+```
+bGFza2RqZmxha3NkamZsa2Fqc2Rsa2ZqYWtsc2RqZmtramRmYWxkc2tm
+```
+
+Then using the `--set-file` flag for OPA
+
+```bash
+opa run --set-file "services.acmecorp.credentials.bearer.token=/var/run/secrets/bearer_token.txt"
+```
+
+It will read the contents of the file and set the config value with the token.
```

Comment:
I found out the hard way that the file loading doesn't strip newlines. As a result, when I tried to override the URL using a file, I was getting url.Parse errors from deep inside the REST client configuration validation logic. It took a few iterations of wrapping returned errors with context to figure out where the problem was.

What if we made `--set-file` strip leading and trailing whitespace? If we run into a situation where whitespace is important we could introduce a `--set-file-raw` variant. WDYT?

Reply:
Good catch. Definitely agree it's more likely that a user wouldn't want trailing whitespace. I'll add that in.